addons/point_of_sale/models/pos_order.py

```diff
@@ -903,36 +955,52 @@ def create_from_ui(self, orders, draft=False):
             if not existing_draft_order:
                 existing_draft_order = self.env['pos.order'].search(['&', ('pos_reference', '=', order['data']['name']), ('state', '=', 'draft')], limit=1)
 
-            if existing_draft_order:
-                order_ids.append(self._process_order(order, draft, existing_draft_order))
-            else:
-                existing_orders = self.env['pos.order'].search([('pos_reference', '=', order['data']['name'])])
-                if all(not self._is_the_same_order(order['data'], existing_order) for existing_order in existing_orders):
-                    order_ids.append(self._process_order(order, draft, False))
+            try:
+                if existing_draft_order:
+                    order_ids.append(self._process_order(order, draft, existing_draft_order))
+                else:
+                    existing_orders = self.env['pos.order'].search([('pos_reference', '=', order['data']['name'])])
+                    if all(not self._is_the_same_order(order['data'], existing_order) for existing_order in existing_orders):
+                        order_ids.append(self._process_order(order, draft, False))
+                _logger.info('The PoS order %s was successfully processed', order['id'])
+            except:
+                self.env.cr.rollback()
```

Comment:
It may roll back others POS orders (from `orders`) that have already been successfully processed. But they won't be committed or saved as attachments (and then you still have the same problem).
Or I miss something ?

Reply:
@ryv-odoo If I am not mistaken, it is actually already the case even without my changes as the exception will rollback the cursor. It will need more test on my end, but it's possible that the PoS send them one by one when a batch of orders fail to sync (but it could be double checked indeed)